--- 
title: "Pathways" 
subtitle: "CBW workshop"
author: "Gary Bader, Veronique Voisin, Ruth Isserlin, Lincoln Stein, Robin Haw, Micheal Hoffman, Quaid Morris"
date: "last modified `r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/bookdown-demo
description: "Course covers the bioinformatics concepts and tools available for interpreting a gene list using pathway and network information. "
---

# Welcome <a id="welcome"></a>

Welcome to Pathways and Network Analysis of -Omics Data 2020. 

Meet your faculty [here](https://drive.google.com/a/bioinformatics.ca/file/d/1-YV2KzEDyaU6opdlXt9Ir6VQSmCCv1DZ/view?usp=sharing).  

**CLASS MATERIALS:** You can download the printed course manual [here](https://drive.google.com/a/bioinformatics.ca/file/d/1HcPuiYUJe69w3_0aNpAfhk7DipcacA6r/view?usp=sharing).  

## Workshop Schedule {#schedule}

| | **Day 1** | | **Day 2** | | **Day 3** |
| :---: | :---: | :---: | :---: | :---: | :---: |
| | **Monday July, 27** | | **Tuesday, July 28** | | **Wednesday, July 29** |
| 8:30 | <font color="green">*Registration and Breakfast*</font> | 8:30 | <font color="green">*Breakfast*</font> | 8:30 | <font color="green">*Breakfast*</font> |
| 9:00 | <font color="purple">Welcome (Michelle Brazas)</font> | | | | |  
| 9:30 | <font color="purple">Module 1: Introduction to Pathway and Network Analysis (Gary Bader)</font> | 9:00 | <font color="purple">Module 4: More Depth on Network and Pathway Analysis (Robin Haw)</font> | 9:00 | <font color="purple">Module 6: Regulatory Network Analysis (Michael Hoffman)</font> |
| 11:00 | <font color="green">*Coffee Break*</font> | 10:00 | <font color="green">*Coffee Break*</font> | 10:30 | <font color="green">*Coffee Break*</font> |
| 11:30 | <font color="purple">Module 2: Finding Over-Represented Pathways (Veronique Voisin)</font> | 10:30 | <font color="blue">Module 4 Lab (Robin)</font> | 11:00 |  <font color="blue">Module 6 Lab (Veronique)</font> |
| 12:30 | <font color="green">*Lunch*</font> | 12:30 | <font color="green">*Lunch*</font> | 12:30 | **Survey and Closing Remarks** |
| 1:30 | <font color="blue">Module 2 Lab (Veronique Voisin)</font> | 1:30 | <font color="purple">Module 5: Gene Function Prediction (Quaid Morris)</font> | 1:00 |  **Integrated Assignment** |
| 3:00 | <font color="green">*Coffee Break*</font> | 3:00 | <font color="green">*Coffee Break*</font> | | |
| 3:30 | <font color="purple">Module 3: Network Visualization and Analysis with Cytoscape (Gary Bader)</font> | 3:30 | <font color="blue">Module 5 Lab (Veronique Voisin)</font> |
| 4:15 | <font color="blue">Module 3 Lab (Veronique Voisin)</font> | | | | |
| 5:00 | <font color="green">*Dinner*</font> | 5:00 | <font color="green">*Dinner*</font> | | |

## Pre-Workshop Materials and Laptop Setup Instructions {#pre-workshop}

### Laptop Setup Instructions

Instructions to setup your laptop can be found ## Install these tools on your laptop before coming to the workshop:

1) A robust text editor.   

* For Windows/PC - [notepad++](http://notepad-plus-plus.org/)  
* For Linux - [gEdit](http://projects.gnome.org/gedit/)  
* For Mac – [TextWrangler](http://www.barebones.com/products/textwrangler/download.html)

2) A file decompression tool.  

* For Windows/PC – [7zip](http://www.7-zip.org/).  
* For Linux – [gzip](http://www.gzip.org).   
* For Mac – already there.

3) A robust internet browser such as Firefox or Safari (Internet Explorer and Chrome are not recommended because of Java issues).

4) Java -The software that we will be using (GSEA and Cytoscape) requires Java. Check if you have Java installed: https://www.java.com/verify/ and download Java if you do not have it installed. **You need Java 8. Do NOT install Java 10**.

5) A PDF viewer (Adobe Acrobat or equivalent).

6) Install [Cytoscape 3.7.1](https://cytoscape.org/download-platforms.html).  

Choose the version corresponding to your operating system (OS, Windows or UNIX) 
Cytoscape requires Java8: check your version at  https://www.java.com/verify/ and download Java8 if you do not have it installed. Contact your system administrator if you have trouble with Java installation. 

7) Within the Cytoscape program, install the following Cytoscape apps.  

From the menu bar, select ‘Apps’, then ‘App Manager’.
 
Within 'all apps', search for the following and install:  

 * EnrichmentMap 3.2.0
 * EnrichmentMap Pipeline Collection 1.0.0 (it will install ClusterMaker2, WordCloud and AutoAnnotate) 
 * GeneMANIA 3.5.1
 * IRegulon  1.3
 * ReactomeFIPlugin 7.2.0 - http://apps.cytoscape.org/apps/reactomefiplugin  
 * stringApp 1.4.2
 * yFiles Layout Algorithms	1.0.2	
 
 
8) Install the data set within GeneMANIA app.

From the menu bar, select 'Apps', hover over 'GeneMANIA', then select 'Choose Another Data Set'.  
From the list of available data sets, select the most recent (2017-07-13/13 July 2017) and under ‘Include only these networks:' select ‘all’. Click on ‘Download’.  
An ‘Install Data' window will pop-up. Select H.Sapiens Human (2589 MB). Click on ‘Install’.  
This requires time and a good network connection to download completely, so be patient (around 15mins).  

  
9) Install GSEA.  

 * Go to the [GSEA page](http://www.broadinstitute.org/gsea/index.jsp)    
 * Register  
 * Login  
 
**option 1 recommended:**
 
 * In menu, in the javaGSEA Desktop Application section, changed *Launch with" to *4GBG (for 64 bit Java only)*  
 * Click on the *Launch* button. This is downloading a gsea icon. Save it on your computer. Double click on this icon to 
 open GSEA.
 
  * Note 1: please try to lauch GSEA before the workshop (open).
  * Note 2: Most common issue with GSEA during the workshop is out of memory error because GSEA was launched with 1GB instead of 4 GB. Please make sure that you have downloaded the 4GB start icon as descrived above.
 
** option 2 in case option 1 does not work on your computer:**
 
 * In menu, choose Downloads 
 * Go to the javaGSEA Java Jar file section and download the gsea-3.0.jar file and save in your Documents folder (do not leave it in the “Downloads”folder).  
 
**MAC/Linux Computer** 

* On a MAC, the Terminal window is located in Applications/Utilities. Tip: add the terminal window to your dock so it is easy to open when needed.  
* At the prompt, type the command in your terminal window and hit enter:

```
java -Xmx4G -jar ~/Documents/gsea-3.0.jar
```

**PC/Windows Computer** 

* On Windows, go to the start icon and type cmd (for command prompt) in the search box.  
* At the prompt, type the following commands, hitting enter in between each command and waiting for the prompt before the next command:

```
cd Documents
java -Xmx4G -jar gsea-3.0.jar
```


### Pre-workshop Tutorials

It is in your best interest to complete these before the workshop.

#### Read the following papers before coming to the workshop:

  * ***Pathway enrichment analysis and visualization of omics data using g:Profiler, GSEA, Cytoscape and EnrichmentMap***<BR>
Reimand J, Isserlin R, Voisin V, Kucera M, Tannus-Lopes C, Rostamianfar A, Wadi L, Meyer M, Wong J, Xu C, Merico D, Bader GD <BR>
[Nat Protoc. 2019 Feb;14(2):482-517](https://www.nature.com/articles/s41596-018-0103-9) - [Available here as well](http://baderlab.org/Publications#EM_2019)
  
  * ***iRegulon: from a gene list to a gene regulatory network using large motif and track collections***<BR>Janky R, Verfaillie A, Imrichová H, Van de Sande B, Standaert L, Christiaens V, Hulselmans G, Herten K, Naval Sanchez M, Potier D, Svetlichnyy D, Kalender Atak Z, Fiers M, Marine JC, Aerts S<BR>[PLoS Comput Biol. 2014 Jul 24;10(7)](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003731) 

  * ***The GeneMANIA prediction server: biological network integration for gene prioritization and predicting gene function***<BR>
Warde-Farley D, Donaldson SL, Comes O, Zuberi K, Badrawi R, Chao P, Franz M, Grouios C, Kazi F, Lopes CT, Maitland A, Mostafavi S, Montojo J, Shao Q, Wright G, Bader GD, Morris Q<BR>
[Nucleic Acids Res 2010 Jul 1;38 Suppl:W214-20](https://academic.oup.com/nar/article/38/suppl_2/W214/1126704) - [Available here as well](http://baderlab.org/Publications#GeneMANIA_original)
  
  * ***GeneMANIA update 2018***<BR>Franz M, Rodriguez H, Lopes C, Zuberi K, Montojo J, Bader GD, Morris Q<BR>
[Nucleic Acids Res. 2018 Jun 15](https://academic.oup.com/nar/article/46/W1/W60/5038280) - [Available here as well](http://baderlab.org/Publications#GeneMANIA_2018)
  
  * ***How to visually interpret biological data using networks***<BR>
Merico D, Gfeller D, Bader GD <BR>
[Nature Biotechnology 2009 Oct 27, 921-924](https://www.nature.com/articles/nbt.1567)  - [Available here as well](http://baderlab.org/Publications#interpret_networks)
  
  * ***g:Profiler--a web-based toolset for functional profiling of gene lists from large-scale experiments.***<BR>Reimand J, Kull M, Peterson H, Hansen J, Vilo J <BR> [Nucleic Acids Res. 2007 Jul;35](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1933153/)
  
  * ***g:Profiler: a web server for functional enrichment analysis and conversions of gene lists (2019 update)***<BR>Raudvere U, Kolberg L, Kuzmin I, Arak T, Adler P, Peterson H, Vilo J<BR>[Nucleic Acids Res. 2019 May 8](https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkz369/5486750)
  
  * ***Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles***<BR>Subramanian A, Tamayo P, Mootha VK, Mukherjee S, Ebert BL, Gillette MA, Paulovich A, Pomeroy SL, Golub TR, Lander ES, Mesirov JP<BR>[Proc Natl Acad Sci U S A. 2005 Oct 25;102(43)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/)
  
  * ***Expression data analysis with Reactome***<BR>Jupe S, Fabregat A, Hermjakob H<BR>[Curr  Protoc Bioinformatics. 2015 Mar 9;49:8.20.1-9](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4407007/)
 
#### Do these exercises before coming to the workshop:

  1. [Sequencing Terminology](http://www.ncbi.nlm.nih.gov/projects/genome/glossary.shtml)

  1.  **Cytoscape Preparation tutorials**: From [introductory tutorial to Cytoscape](http://tutorials.cytoscape.org), complete:
  
  * 1. [Tour of Cytoscape](https://cytoscape.org/cytoscape-tutorials/protocols/tour-of-cytoscape/#/)
  * 2. [Basic Data Visualization](https://cytoscape.org/cytoscape-tutorials/protocols/basic-data-visualization/#/)

***


### Additional tutorials <a id="additionaltutorials"></a>

EnrichmentMap (GSEA and g:Profiler) step by step protocol:
https://www.biorxiv.org/content/early/2017/12/12/232835
and https://github.com/BaderLab/Cytoscape_workflows

Interacting with Cytoscape using CyRest and command lines (for advanced users):
https://github.com/cytoscape/cytoscape-automation/blob/master/for-scripters/R/advanced-cancer-networks-and-data-rcy3.Rmd


***

Thank you for attending the Pathway and Network Analysis of Omics Data workshop! Help us make this workshop better by filling out [our survey](https://forms.gle/D8w8qyJ1r71rFnZe9).

*** 

```{r include=FALSE}
# automatically create a bib database for R packages
knitr::write_bib(c(
  .packages(), 'bookdown', 'knitr', 'rmarkdown'
), 'packages.bib')
```

Icons are from the [“Very Basic. Android L Lollipop” set by Ivan Boyko](https://www.iconfinder.com/iconsets/very-basic-android-l-lollipop) licensed under [CC BY 3.0](https://creativecommons.org/licenses/by/3.0/). 
